Node/src/RemoteQuery/teams.js

```diff
@@ -437,4 +438,130 @@ Teams.prototype.fetchTeamInfo = function (teamsId, options, callback) {
   });
 }
 
+/**
+ * @summary Begin a new reply chain in channel
+ *
+ * Begin a new reply chain in channel.
+ *
+ * @param {string} channelId Channel Id
+ *
+ * @param {object} message to send in channel.
+ *
+ * @param {object} [options] Optional Parameters.
+ *
+ * @param {object} [options.customHeaders] Headers that will be added to the
+ * request
+ *
+ * @param {function} callback - The callback.
+ *
+ * @returns {function} callback(err, result, request, response)
+ *
+ *                      {Error}  err        - The Error object if an error occurred, null otherwise.
+ *
+ *                      {object} [result]   - The deserialized result object if an error did not occur.
+ *                      See {@link IChatConnectorAddress} for more information.
+ *
+ *                      {object} [request]  - The HTTP Request object if an error did not occur.
+ *
+ *                      {stream} [response] - The HTTP Response stream if an error did not occur.
+ */
+
+Teams.prototype.beginReplyChainInChannel = function (channelId, message, options, callback) {
+   /* jshint validthis: true */
+  let client = this.client;
+  if(!callback && typeof options === 'function') {
+    callback = options;
+    options = null;
+  }
+  if (!callback) {
+    throw new Error('callback cannot be null.');
+  }
+  // Validate
+  try {
+    if (channelId === null || channelId === undefined || typeof channelId.valueOf() !== 'string') {
+      throw new Error('channelId cannot be null or undefined and it must be of type string.');
+    }
+  } catch (error) {
+    return callback(error);
+  }
+
+  // Construct URL
+  let baseUrl = this.client.baseUri;
+  let requestUrl = baseUrl + (baseUrl.endsWith('/') ? '' : '/') + 'v3/conversations';
+
+  // Create HTTP transport objects
+  let httpRequest = new WebResource();
+  options.method = 'POST';
+  options.url = requestUrl;
+  options.headers = options['customHeaders'];
+  // Set Body
+  options.body = {
+    activity: message,
+    channelData: {
+      teamsChannelId: channelId
+    },
+    json: true
+  };
+  httpRequest.prepare(options);
+  // Set Headers
+  httpRequest.headers['Content-Type'] = 'application/json; charset=utf-8';
+  // Send Request
+  return client.pipeline(httpRequest, (err, response, responseBody) => {
+    if (err) {
+      return callback(err);
+    }
+    let statusCode = response.statusCode;
+    if (statusCode !== 201) {
+      let error = new Error(responseBody);
```

Comment:
Should we refactor this? There are a couple of places in this file which use the same logic.